RELEASES.md

```diff
@@ -1,3 +1,440 @@
+
+In the future we plan on providing some more ink! workshops and tutorials guiding the approach to design and implement a custom storage data structure.
+
+### In Summary
+
+The new `ink_storage` crate provides everything you need to operate on your contract's storage.
+There are low-level and high-level data structures depending on your need of control.
+All provided data structures operate lazily on the contract's storage and cache their reads and writes for a more gas efficient storage access.
+Users should prefer high-level data structures found in the `collections` module over the low-level data structures found in the `lazy` module.
+For a list of all the new storage data structure visit [`ink_storage`'s documentation](https://paritytech.github.io/ink/ink_storage/).
+
+## ink! Attributes
+
+For ink! 3.0 we have added some more useful ink! specific attributes to the table.
+All of these ink! attributes are available to specify inside an ink! module.
+An ink! module is the module that is flagged by `#[ink::contract]` containing all the ink! definitions:
+```rust
+use ink_lang as ink;
+
+#[ink::contract]
+mod erc20 {
+    #[ink(storage)]
+    pub struct Erc20 { ... }
+
+    impl Erc20 {
+        #[ink(constructor)]
+        pub fn new(initial_supply: Balance) -> Self { .. }
+
+        #[ink(constructor)]
+        pub fn total_supply(&self) -> Balance { .. }
+
+        // etc. ...
+    }
+}
+```
+
+We won't be going into the details for any of those but will briefly present the entire set of ink! specific attributes below:
+
+| Attribute | Where Applicable | Description |
+|:--|:--|:--|
+| `#[ink(storage)]` | On `struct` definitions. | Defines the ink! storage struct. There can only be one ink! storage definition per contract. |
+| `#[ink(event)]` | On `struct` definitions. | Defines an ink! event. A contract can define multiple such ink! events. |
+| `#[ink(anonymous)]` **new** | Applicable to ink! events. | Tells the ink! codegen to treat the ink! event as anonmyous which omits the event signature as topic upon emitting. Very similar to anonymous events in Solidity. |
+| `#[ink(topic)]` | Applicate on ink! event field. | Tells the ink! codegen to provide a topic hash for the given field. Every ink! event can only have a limited number of such topic field. Similar semantics as to indexed event arguments in Solidity. |
+| `#[ink(message)]` | Applicable to methods. | Flags a method for the ink! storage struct as message making it available to the API for calling the contract. |
+| `#[ink(constructor)]` | Applicable to method. | Flags a method for the ink! storage struct as constructor making it available to the API for instantiating the contract. |
+| `#[ink(payable)]` **new** | Applicable to ink! messages. | Allow transactions while calling this ink! message. ink! constructors are implicitely payable. |
```

Comment:
Can be more specific than "transaction", something about receiving value as part of the call

Reply:
done